addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveProtectionCommandClass.java

```diff
@@ -64,10 +111,243 @@ public void handleApplicationCommandRequest(SerialMessage serialMessage, int off
+                SerialMessageType.Request, SerialMessageClass.ApplicationCommandHandler, SerialMessagePriority.Get);
+
+        ByteArrayOutputStream outputData = new ByteArrayOutputStream();
+        outputData.write(getNode().getNodeId());
+        outputData.write(2);
+        outputData.write(getCommandClass().getKey());
+        outputData.write(PROTECTION_GET);
+
+        result.setMessagePayload(outputData.toByteArray());
+        return result;
+    }
+
+    /**
+     * Gets a SerialMessage with the PROTECTION_SET command
+     *
+     * @return the serial message, or null if the supported command is not supported.
+     */
+    public SerialMessage setValueMessage(LocalProtectionType localMode, RfProtectionType rfMode) {
+        logger.debug("NODE {}: Creating new message for command PROTECTION_SET", getNode().getNodeId());
+
+        SerialMessage result = new SerialMessage(getNode().getNodeId(), SerialMessageClass.SendData,
+                SerialMessageType.Request, SerialMessageClass.SendData, SerialMessagePriority.Set);
+
+        LocalProtectionType newLocalMode = localMode != null ? localMode : currentLocalMode;
+
+        ByteArrayOutputStream outputData = new ByteArrayOutputStream();
+        if (getVersion() < 2 || rfMode == null) {
+            outputData.write(getNode().getNodeId());
+            outputData.write(3);
+            outputData.write(getCommandClass().getKey());
+            outputData.write(PROTECTION_SET);
+            outputData.write(newLocalMode.getKey());
+        } else {
+            outputData.write(getNode().getNodeId());
+            outputData.write(4);
+            outputData.write(getCommandClass().getKey());
+            outputData.write(PROTECTION_SET);
+            outputData.write(newLocalMode.getKey());
+            outputData.write(rfMode.getKey());
+
+        }
+        result.setMessagePayload(outputData.toByteArray());
+        return result;
+    }
+
+    @Override
+    public Collection<SerialMessage> initialize(boolean refresh) {
+        ArrayList<SerialMessage> result = new ArrayList<>();
+        if (getVersion() < 2) {
+            return result;
+        }
+
+        if (refresh == true || supportedInitialised == false) {
+            result.add(getSupportedMessage());
+        }
+        return result;
+    }
+
+    @Override
+    public Collection<SerialMessage> getDynamicValues(boolean refresh) {
+        ArrayList<SerialMessage> result = new ArrayList<>();
+        if (refresh == true || dynamicDone == false) {
+            result.add(getValueMessage());
+        }
+
+        return result;
+    }
+
+    /**
+     * Reported type of protection.
+     */
+    public enum Type {
+        PROTECTION_LOCAL,
+        PROTECTION_RF
+    }
+
+    /**
+     * Z-Wave LocalProtectionType enumeration. The LocalProtection type indicates the type of local protection that is
+     * reported.
+     *
+     * @author Jorg de Jong
+     */
+    @XStreamAlias("localProtection")
```

Comment:
Do we need these 'complex' enums? The label isn't used anywhere (is it) so why not just use a standard enum without the label?
I also wonder if the values shouldn't be uppercase...